src/plugins/pol/polformat.cpp

```diff
@@ -20,320 +20,236 @@
 
 #include "polformat.h"
 
-#include "pregdata.h"
-#include "pregparser.h"
-#include "pregwriter.h"
-
 #include "../../../src/plugins/administrative_templates/registry/registry.h"
 #include "../../../src/plugins/administrative_templates/registry/registryentry.h"
 #include "../../../src/plugins/administrative_templates/registry/registryentrytype.h"
 
+#include "./parser.h"
+
 #include <byteswap.h>
+#include <tuple>
 
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wold-style-cast"
 
 using namespace model::registry;
 
-namespace gpui
-{
+namespace gpui {
 class RegistryEntryAdapter
 {
-private:
-    static std::unique_ptr<model::registry::AbstractRegistryEntry> adaptCharEntry(
-        const preg::PregEntry &entry, model::registry::RegistryEntryType type)
+public:
+    static void addInstruction(pol::PolicyTree &tree,
+                               const std::unique_ptr<model::registry::AbstractRegistryEntry> &entry)
     {
-        auto registryEntry   = std::make_unique<model::registry::RegistryEntry<QString>>();
-        registryEntry->key   = entry.key.c_str();
-        registryEntry->type  = type;
-        registryEntry->value = entry.value.c_str();
-        if (entry.data)
-        {
-            void *pt            = entry.data;
-            size_t requiredSize = entry.size;
-            registryEntry->data = QString::fromUtf16(
-                reinterpret_cast<char16_t *>(std::align(alignof(char16_t), sizeof(char), pt, requiredSize)));
-            delete[] entry.data;
-        }
+        pol::PolicyInstruction instruction;
 
-        return registryEntry;
-    }
+        auto key = entry->key.toStdString();
+        auto value = entry->value.toStdString();
 
-    static std::unique_ptr<model::registry::AbstractRegistryEntry> adaptUInt32Entry(
-        const preg::PregEntry &entry, model::registry::RegistryEntryType type, bool bigEndian = false)
-    {
-        auto registryEntry   = std::make_unique<model::registry::RegistryEntry<uint32_t>>();
-        registryEntry->key   = entry.key.c_str();
-        registryEntry->type  = type;
-        registryEntry->value = entry.value.c_str();
-        if (entry.data)
-        {
-            uint32_t data = (uint32_t)((uint8_t) entry.data[0]) | (uint32_t)((uint8_t) entry.data[1]) << 8
-                            | (uint32_t)((uint8_t) entry.data[2]) << 16 | (uint32_t)((uint8_t) entry.data[3]) << 24;
-
-            if (bigEndian)
-            {
-                data = bswap_32(data);
-            }
-            registryEntry->data = data;
-            delete[] entry.data;
+        if (tree.find(key) == tree.end()) {
+            tree[key] = {};
+        }
+        if (tree[key].find(value) != tree[key].end()) {
+            qWarning() << "Duplicate policy entry with key: " << entry->key
+                       << ", value:" << entry->value;
+            return;
```

Comment:
Why do we have to return here it is common case to have multiple values with same key and value fields.